gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergPartitionDatasetTest.java

```diff
@@ -225,39 +199,57 @@ public void testWithDifferentSrcAndDestTableWriteLocation() throws IOException {
     List<CopyEntity> copyEntities =
         (List<CopyEntity>) icebergPartitionDataset.generateCopyEntities(targetFs, copyConfiguration);
 
-    Assert.assertEquals(copyEntities.size(), 2);
-    verifyCopyEntities(copyEntities, false);
+    verifyCopyEntities(copyEntities, 2, false);
   }
 
-  private List<DataFile> getDataFiles() throws IOException {
+  private static void setupSrcFileSystem() throws IOException {
+    sourceFs = Mockito.mock(FileSystem.class);
+    Mockito.when(sourceFs.getUri()).thenReturn(SRC_FS_URI);
+    Mockito.when(sourceFs.makeQualified(any(Path.class)))
+        .thenAnswer(invocation -> invocation.getArgument(0, Path.class).makeQualified(SRC_FS_URI, new Path("/")));
+    Mockito.when(sourceFs.getFileStatus(any(Path.class))).thenAnswer(invocation -> {
+      Path path = invocation.getArgument(0, Path.class);
+      Path qualifiedPath = sourceFs.makeQualified(path);
+      return IcebergDatasetTest.MockFileSystemBuilder.createEmptyFileStatus(qualifiedPath.toString());
+    });
+  }
+
+  private static void setupDestFileSystem() throws IOException {
+    targetFs = Mockito.mock(FileSystem.class);
+    Mockito.when(targetFs.getUri()).thenReturn(DEST_FS_URI);
+    Mockito.when(targetFs.makeQualified(any(Path.class)))
+        .thenAnswer(invocation -> invocation.getArgument(0, Path.class).makeQualified(DEST_FS_URI, new Path("/")));
+    // Since we are adding UUID to the file name for every file while creating destination path,
+    // so return file not found exception if trying to find file status on destination file system
+    Mockito.when(targetFs.getFileStatus(any(Path.class))).thenThrow(new FileNotFoundException());
+  }
+
+  private static List<DataFile> createDataFileMocks() throws IOException {
     List<DataFile> dataFiles = new ArrayList<>();
     for (String srcFilePath : srcFilePaths) {
       DataFile dataFile = Mockito.mock(DataFile.class);
       Path dataFilePath = new Path(srcFilePath);
       Path qualifiedPath = sourceFs.makeQualified(dataFilePath);
       Mockito.when(dataFile.path()).thenReturn(dataFilePath.toString());
-      Mockito.when(sourceFs.getFileStatus(Mockito.eq(dataFilePath))).thenReturn(getFileStatus(qualifiedPath));
+      Mockito.when(sourceFs.getFileStatus(Mockito.eq(dataFilePath))).thenReturn(
+          IcebergDatasetTest.MockFileSystemBuilder.createEmptyFileStatus(qualifiedPath.toString()));
       dataFiles.add(dataFile);
     }
     return dataFiles;
   }
 
-  private static FileStatus getFileStatus(Path path) {
-    FileStatus fileStatus = new FileStatus();
-    fileStatus.setPath(path);
-    return fileStatus;
-  }
-
-  private static void verifyCopyEntities(Collection<CopyEntity> copyEntities, boolean sameSrcAndDestWriteLocation) {
+  private static void verifyCopyEntities(Collection<CopyEntity> copyEntities, int expectedCopyEntitiesSize,
```

Comment:
taking in the expected size is a definite improvement, but let's go further to also take in the `expectedSrcFilePaths` - please [see my reply](https://github.com/apache/gobblin/pull/4058#discussion_r1811938323)

Reply:
Done